nuxeo-drive-client/nxdrive/engine/engine.py

```diff
@@ -901,14 +901,15 @@ def _add_top_level_state(self):
 
     def suspend_client(self, reason):
         if self.is_paused() or self._stopped:
-            raise ThreadInterrupt
+            raise ThreadInterrupt("[Engine.suspend_client] client paused or stopped: %s" % repr(reason))
         # Verify thread status
         thread_id = current_thread().ident
         for thread in self._threads:
             if hasattr(thread, "worker") and isinstance(thread.worker, Processor):
                 if (thread.worker._thread_id == thread_id and
                         thread.worker._continue == False):
-                    raise ThreadInterrupt
+                    raise ThreadInterrupt("[Engine.suspend_client] Worker thread %s _continue flag is False: %s" %
+                                          (thread_id, repr(reason)))
```

Comment:
I suspect the message part of ThreadInterrupt is not logged for this scenario because the Processor catches ThreadInterrupt but does not process the message part of it (like writing it to log file).